_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -909,40 +916,128 @@ public static function jetpack_connection_status() {
 	}
 
 	/**
-	 * Test connection status for this Jetpack site. It uses the /jetpack-blogs/%d/test-connection wpcom endpoint.
+	 * Test connection status for this Jetpack site.
 	 *
 	 * @since 6.8.0
 	 *
 	 * @return array|WP_Error WP_Error returned if connection test does not succeed.
 	 */
 	public static function jetpack_connection_test() {
-		$response = Jetpack_Client::wpcom_json_api_request_as_blog(
-			sprintf( '/jetpack-blogs/%d/test-connection', Jetpack_Options::get_option( 'id' ) ),
-			Jetpack_Client::WPCOM_JSON_API_VERSION
+		jetpack_require_lib( 'debugger' );
+		$cxntests = new Jetpack_Cxn_Tests();
+
+		if ( $cxntests->pass() ) {
+			return rest_ensure_response(
+				array(
+					'code'    => 'success',
+					'message' => __( 'All connection tests passed.', 'jetpack' ),
+				)
+			);
+		} else {
+			return $cxntests->output_fails_as_wp_error();
+		}
+	}
+
+	/**
+	 * Test connection permission check method.
+	 *
+	 * @since 6.9.0
+	 *
+	 * @return bool
+	 */
+	public static function view_jetpack_connection_test_check() {
+		$signature = base64_decode( $_GET['signature'] );
```

Comment:
We should probably verify `$_GET` before we do anything here. I get the following notices when loading the Debug page on a connected site:

```
[29-Jan-2019 09:17:32 UTC] PHP Notice:  Undefined index: signature in /wp-content/plugins/jetpack-dev/_inc/lib/class.core-rest-api-endpoints.php on line 949
[29-Jan-2019 09:17:32 UTC] PHP Notice:  Undefined index: timestamp in /wp-content/plugins/jetpack-dev/_inc/lib/class.core-rest-api-endpoints.php on line 954
[29-Jan-2019 09:17:32 UTC] PHP Notice:  Undefined index: url in /wp-content/plugins/jetpack-dev/_inc/lib/class.core-rest-api-endpoints.php on line 955
```

Reply:
I'd like to revert this check for now until I better understand the need it is trying to solve. The encoding of the result should be sufficient and the wpcom side wasn't set to send anything when requesting it, so this in effect seemingly breaks the functionality.

Reply:
Added it back and added isset checks.